source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -516,10 +541,39 @@ namespace
         }
     }
 
-    void writeDocument(DocumentPtr& doc, PrvStage* stage, RtFileIo::WriteFilter filter)
+    void writeSourceUris(PrvStage* stage, DocumentPtr doc)
+    {
+        const PrvObjectHandleVec& refs = stage->getReferencedStages();
+        for (size_t i = 0; i < refs.size(); i++)
+        {
+            const PrvObjectHandle refStage = refs[i];
+            PrvStage* pStage = refStage->asA<PrvStage>();
```

Comment:
Nitpick: Could be merged into one line to avoid the extra refcount increment/decrement when copying the PrvObjectHandle:

`const PrvStage* refStage = refs[i]->asA<PrvStage>();`